signer/tests/integration/postgres.rs

```diff
@@ -4919,3 +4919,938 @@ async fn pending_rejected_withdrawal_already_accepted() {
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(
+                bitcoin_chain_tip.as_ref(),
+                &stacks_chain_tip,
+                3,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+        assert_eq!(requests.len(), 1);
+
+        // Context window 2, the request should NOT be returned.
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(
+                bitcoin_chain_tip.as_ref(),
+                &stacks_chain_tip,
+                2,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+        assert!(requests.is_empty());
+    }
+
+    /// Asserts that requests with a confirmed rejection event are not returned.
+    ///
+    /// This test creates blockchains with the following structure:
+    ///
+    /// ```text
+    ///          ┌────────┐
+    /// Bitcoin: │   B1   │
+    ///          └─▲──────┘  We both confirm (✔) and reject (✖) the
+    ///          ┌─┴──────┐  request in S1.
+    /// Stacks:  │  S1 ✔✖ │
+    ///          └────────┘
+    /// ```
+    #[tokio::test]
+    async fn request_with_confirmed_rejection_not_returned() {
+        let db = signer::testing::storage::new_test_database().await;
+
+        let signature_threshold = 2;
+        let context_window = 1_000;
+
+        // Bitcoin blocks:
+        let bitcoin_block = BitcoinBlock::new_genesis();
+        // Stacks blocks:
+        let stacks_block = StacksBlock::new_genesis().anchored_to(&bitcoin_block);
+
+        // Write our blocks.
+        db.write_blocks([&bitcoin_block], [&stacks_block]).await;
+
+        // Store a withdrawal request.
+        let request =
+            store_withdrawal_request(&db, 1, &bitcoin_block, &stacks_block, &[true, true]).await;
+
+        // We should get the request as pending accepted.
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(
+                &bitcoin_block.block_hash,
+                &stacks_block.block_hash,
+                context_window,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+        assert_eq!(requests.len(), 1);
+
+        // Now reject the request.
+        reject_withdrawal_request(&db, &request, &stacks_block).await;
+
+        // It should no longer be returned.
+        let requests = db
+            .get_pending_accepted_withdrawal_requests(
+                &bitcoin_block.block_hash,
+                &stacks_block.block_hash,
+                context_window,
+                signature_threshold,
+            )
+            .await
+            .expect("failed to query db");
+        assert!(requests.is_empty());
+    }
+
+    /// Asserts that a withdrawal request is returned when it has been confirmed
+    /// and has a rejection event, but the rejection event has been orphaned.
```

Comment:
Hopefully we never have to deal with this case. If it happens then we did something wrong. Glad we have the test though.

Reply:
Yeah, it's less about testing though and more about asserting the behavior of the query given the case.